application/models/organizationtrust.go

```
@@ -0,0 +1,121 @@
+	"github.com/gobuffalo/validate"
+	"github.com/gobuffalo/validate/validators"
+
+	"github.com/silinternational/wecarry-api/domain"
+)
+
+// OrganizationTrust is the model for storing Organization connections, also known as OrganizationTrusts
+type OrganizationTrust struct {
+	ID          int       `json:"id" db:"id"`
+	CreatedAt   time.Time `json:"created_at" db:"created_at"`
+	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
+	PrimaryID   int       `json:"primary_id" db:"primary_id"`
+	SecondaryID int       `json:"secondary_id" db:"secondary_id"`
+}
+
+// OrganizationTrusts is used for struct-attached functions that operate on lists of objects
+type OrganizationTrusts []OrganizationTrust
+
+// Validate gets run every time you call a "pop.Validate*" (pop.ValidateAndSave, pop.ValidateAndCreate, pop.ValidateAndUpdate) method.
+func (o *OrganizationTrust) Validate(tx *pop.Connection) (*validate.Errors, error) {
+	return validate.Validate(
+		&validators.IntIsPresent{Field: o.PrimaryID, Name: "PrimaryID"},
+		&validators.IntIsPresent{Field: o.SecondaryID, Name: "SecondaryID"},
+		&validators.IntsAreNotEqual{ValueOne: o.PrimaryID, ValueTwo: o.SecondaryID, Name: "SecondaryEqualsPrimary"},
+	), nil
+}
+
+// ValidateCreate gets run every time you call "pop.ValidateAndCreate" method.
+func (o *OrganizationTrust) ValidateCreate(tx *pop.Connection) (*validate.Errors, error) {
+	return validate.NewErrors(), nil
+}
+
+// ValidateUpdate gets run every time you call "pop.ValidateAndUpdate" method.
+func (o *OrganizationTrust) ValidateUpdate(tx *pop.Connection) (*validate.Errors, error) {
+	return validate.NewErrors(), nil
+}
+
+// CreateSymmetric creates two records to make a two-way trust connection
+func (o *OrganizationTrust) CreateSymmetric() error {
+	if err := o.Create(); err != nil {
+		return err
+	}
+
+	mirror := OrganizationTrust{
+		PrimaryID:   o.SecondaryID,
+		SecondaryID: o.PrimaryID,
+	}
+	if err := mirror.Create(); err != nil {
+		_ = DB.Destroy(o)
```

Maybe return the error from here?